src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
+        size++;
+        putValue(key, value);
+    }
+
+    private void putValue(K key, V value) {
+        int hash = hash(key);
+        int index;
+        Node<K, V> node;
+        Node<K, V> oldNode;
+
+        if (table == null || table.length == 0) {
+            resize();
+        }
```

Comment:
why do we need resize if table is empty? Lets create constructor and initialize table with default capacity in it